### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by ShiroYaksha90**
## Hi 👋 , in this pull request we:
- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. 
- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

**Review by r-ahmed2022 — COMMENTED** [review]
yes, implement changes

File: config/routes.rb [hunk 1]
```diff
@@ -3,13 +3,21 @@
...
     end
   end
-  root "users#index"
 
+  namespace :api, defaults: { format: :json } do
+    namespace :v1 do
+      resources :users, only: [:index, :show] do
+        resources :posts, only: [:index, :show] do
+          resources :comments, only: [:index, :create]
+        end
```
Comment (Owusu-Desmond) [inline config/routes.rb:19]: - [ ] Based on the project requirements, there needs to be an API endpoint that retrieves all comments for a post and posts comments on a post. The endpoint should be accessed using a GET and POST request to `/api/v1/users/user_id/posts/post_id/comments`. However, when I attempted to access this endpoint to get comments and post comments, I got this error message below :point_down: . Please make sure that you fix this issue.

  
  Screenshot from 2023-07-29 18-43-30
  
  image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @ShiroYaksha90 and @r-ahmed2022 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ShiroYaksha90** [comment]
Hi @Owusu-Desmond , please check again as I failed to upload the comments_controller file, but now it should work
comments
com
json

**Comment by Owusu-Desmond** [comment]
Hi @ShiroYaksha90, Submit the project again for re-review since you didn't  commit the latest commit earlier and due to that I didn't get the up-to-date changes

**Comment by ShiroYaksha90** [comment]
@Owusu-Desmond Yes sorry about that.
I re-submitted it as you instructed

File: config/routes.rb (hunk 1 above)
Comment (peterrobert) [reply config/routes.rb:19]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @ShiroYaksha90, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ShiroYaksha90** [comment]
@Owusu-Desmond @peterrobert Thank you both and have a pleasant night

**Pull request merged**